File: ProyectoMysql/server/EntityLayer/RecepcionDetalleEntity.py
from datetime import datetime
from pydantic import BaseModel
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum


class RecepcionDetalleSaveModel(BaseModel):
    RecepcionDetalleId: int
    RecepcionId: int
    MercaderiaId: int
    Cantidad: float
    Lote: str
    FechaIngreso: datetime
    FechaRegistro: datetime
    FechaVencimiento: datetime
    Observacion: str
    Action: ProcessActionEnum


class RecepcionDetalleItemModel:
    RecepcionDetalleId: int
    RecepcionId: int
    MercaderiaId: int
    Cantidad: float
    Lote: str
    FechaIngreso: datetime
    FechaRegistro: datetime
    Observacion: str
    FechaVencimiento: datetime
    NomProducto: str
    CodigoUM: str

    def CargarItem(_DB):
        c = RecepcionDetalleItemModel()
        c.RecepcionDetalleId = _DB["RecepcionDetalleId"]
        c.RecepcionId = _DB["RecepcionId"]
        c.MercaderiaId = _DB["MercaderiaId"]
        c.Cantidad = _DB["Cantidad"]
        c.Lote = _DB['Lote']
        c.FechaIngreso = _DB["FechaIngreso"]
        c.FechaRegistro = _DB["FechaRegistro"]
        c.Observacion = _DB["Observacion"]
        c.FechaVencimiento = _DB["FechaVencimiento"]
        c.NomProducto = _DB["NomProducto"]
        c.CodigoUM = _DB["CodigoUM"]
        return c

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/RecepcionDetalleEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class RecepcionDetalleEntity :BaseEntityObject
    {
        public RecepcionDetalleEntity()
        {

            this.RecepcionDetalleId = 0;
            this.RecepcionId = 0;
            this.MercaderiaId = 0;
            this.Cantidad = 0;
            this.Lote = String.Empty;
            this.FechaIngreso = DateTime.Now;
            this.FechaRegistro = DateTime.Now;
            this.FechaVencimiento = DateTime.Now;
            this.Observacion = String.Empty;
        }
        public Int32 RecepcionDetalleId { get; set; }
        public Int32 RecepcionId { get; set; }
        public Int32 MercaderiaId { get; set; }
        public Decimal Cantidad { get; set; }
        public String Lote { get; set; }
        public DateTime FechaIngreso { get; set; }
        public DateTime FechaRegistro { get; set; }
        public DateTime FechaVencimiento { get; set; }
        public String Observacion { get; set; }

    }
}
